### ISSUE TITLE
How do I stop Karabiner from hijacking my function keys?

### ISSUE THREAD

**Issue by Kalabasa** [pqrs-org/Karabiner-Elements#2374]
I don't need function key mapping. How do I let function keys pass through unnmodified? Currently it's changing Fn+F1 into a Brightness command. Can it output Fn+F1 as Fn+F1?

**Comment by ssbarnea** [comment]
Not sure if we are facing the same issue but on my Mac I am unable to convince Drevo BladeMaster keyboard to use Fn keys instead of multimedia ones. The only way to access function keys was by pressing Fn on the keyboard.

I tried System Settings and also Karabiner options that are supposed to change behavior but none had any effect.

**Comment by MuhammedZakir** [comment]
> I don't need function key mapping. How do I let function keys pass through unnmodified? Currently it's changing Fn+F1 into a Brightness command. Can it output Fn+F1 as Fn+F1?

In `System Preferences > Keyboard`, uncheck `Use F1, F2, etc. keys as standard function keys` if it's checked.

---
@ssbarnea: That's strange! I have enabled above option and I am able to use function keys without 'Fn' on both internal and external keyboards. Which OS are you using? Karabiner does not seem to work on latest MacOS. I am using Karabiner-Elements 12.10.1 on Mojave, btw.

**Comment by ssbarnea** [comment]
I was able to figure it out: I had to disable the MacOS compatibility Sp switch off on my Drevo BladeMaster keyboard in order to make the function keys work while using System Preferences to perform the modifier swap. I find it bit silly from their point of view because they did not document it.

**Comment by paztis** [comment]
why not to propose a way to disable this feature ?
on my touchbar, I want to use the fn key to access to my function keys
on my external keyboard it's the invert (I've no fn key)

This feature set for both the same behavior...

**Comment by MuhammedZakir** [comment]
> on my touchbar, I want to use the fn key to access to my function keys
> on my external keyboard it's the invert (I've no fn key)
> 
> This feature set for both the same behavior...

For now, you can achieve it with complex modifications using `device_if/device_unless` conditions. [1]

[1] https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/conditions/device/

---

> why not to propose a way to disable this feature ?

This is actually a system feature, not Karabiner specific feature. However, as we are able to (kind of) change it per keyboard with complex modifications, It would be great if there is a native feature in KE to do that.

**Comment by lornally** [comment]
> why not to propose a way to disable this feature ? on my touchbar, I want to use the fn key to access to my function keys on my external keyboard it's the invert (I've no fn key)
> 
> This feature set for both the same behavior...

we need this too.

**Comment by ladusee** [comment]
> why not to propose a way to disable this feature ? on my touchbar, I want to use the fn key to access to my function keys on my external keyboard it's the invert (I've no fn key)
 > This feature set for both the same behavior...


Indeed. It would be nice to have an option to disable F-row key modifications, or at least an option "Do not modify events" for individual keys.
